internal/core/src/segcore/SegmentInterface.cpp

```diff
@@ -101,7 +102,8 @@ SegmentInternalInterface::Retrieve(tracer::TraceContext* trace_ctx,
     auto results = std::make_unique<proto::segcore::RetrieveResults>();
     query::ExecPlanNodeVisitor visitor(*this, timestamp);
     auto retrieve_results = visitor.get_retrieve_result(*plan->plan_node_);
-    retrieve_results.segment_ = (void*)this;
+    retrieve_results.segment_ = static_cast<void*>(
+        const_cast<milvus::segcore::SegmentInternalInterface*>(this));
```

Comment:
const_cast?

Reply:
Same as above
**const_cast** to remove the const-ness from `this`
**static_cast** to convert the pointer to `void* `